### DISCUSSION TITLE
The `defaults` top-level key in a workflow should be more useful

### DISCUSSION THREAD

**Discussion by wheelerlaw** [community/community#57462] [Actions]
### Select Topic Area

Product Feedback

### Body

Here is the description in the documentation for the top-level `defaults` key in the workflow syntax:

> Use `defaults` to create a `map` of default settings that will apply to all jobs in the workflow. You can also set default settings that are only available to a job. For more information, see [`jobs.<job_id>.defaults`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_iddefaults).
> 
> When more than one default setting is defined with the same name, GitHub uses the most specific default setting. For example, a default setting defined in a job will override a default setting that has the same name defined in a workflow.

Based on this description, one might reasonably surmise that the following should work to set the default timeout for all jobs in a workflow:

```yaml
name: CI

on:
  pull_request:
    branches:
      - main
  push:
    branches:
      - main

defaults:
  timeout-minutes: 120

jobs:
  ...
```

However, it doesn't. Instead, it results in a syntax error:

```
The workflow is not valid. .github/workflows/CI.yml (Line: 14, Col: 3): Unexpected value 'timeout-minutes'
```

The `defaults` key doesn't seem to actually work for any job-level configuration other than `run`, which seems to contradict the description of the `defaults` key which implies that it is *not* just for the `run` sub-key. 

I'd like to see either the documentation cleaned up for the `defaults` key to indicate that it does not actually provide default setting***s*** (just one setting: `run`) or the `default` key fulfill its described purpose and provide defaults for all job-level settings.

**Comment by Th3S4mur41** [comment]
+1 on this...
I wanted use `defaults` to set `runs-on` and `container` since the value is rarely changed between jobs.
Unfortunately, no luck...

That would be a great addition though and would simplify some workflows substantially
